app/models/reservation.rb

```diff
@@ -67,28 +83,42 @@ def self.number_overdue_for_eq_model(model_id, reservations)
     # 0 queries
     count = 0
     reservations.each do |r|
-      count += 1 if r.status == 'overdue' && r.equipment_model_id == model_id
+      count += 1 if r.overdue && r.equipment_model_id == model_id
     end
     count
   end
 
   ## Getter style instance methods ##
 
-  def status # rubocop:disable CyclomaticComplexity, PerceivedComplexity
-    return if due_date.nil?
-    if checked_out.nil?
-      if approval_status == 'auto' || approval_status == 'approved'
-        due_date >= Time.zone.today ? 'reserved' : 'missed'
-      elsif approval_status
-        approval_status
+  def approved?
+    self.flagged?(:request) && (status != 'denied')
```

Comment:
nice fix, I didn't notice that it was going to return false if it was anything other than `reserved` before.